DataFormats/CaloRecHit/interface/CaloRecHit.h

```diff
@@ -4,27 +4,52 @@
 #include "DataFormats/DetId/interface/DetId.h"
 #include <ostream>
 
+#ifdef __CUDA_ARCH__
+  __constant__
+#else
+  constexpr
+#endif
+  uint32_t calo_rechit_masks[] = {
+    0x00000000u,0x00000001u,0x00000003u,0x00000007u,0x0000000fu,0x0000001fu,
+    0x0000003fu,0x0000007fu,0x000000ffu,0x000001ffu,0x000003ffu,0x000007ffu,
+    0x00000fffu,0x00001fffu,0x00003fffu,0x00007fffu,0x0000ffffu,0x0001ffffu,
+    0x0003ffffu,0x0007ffffu,0x000fffffu,0x001fffffu,0x003fffffu,0x007fffffu,
+    0x00ffffffu,0x01ffffffu,0x03ffffffu,0x07ffffffu,0x0fffffffu,0x1fffffffu,
+    0x3fffffffu,0x7fffffffu,0xffffffffu};
 
 /** \class CaloRecHit
  * 
  *\author J. Mans - Minnesota
  */
 class CaloRecHit {
 public:
-  CaloRecHit(); // for persistence
-  explicit CaloRecHit(const DetId& id, float energy, float time, uint32_t flags = 0, uint32_t aux=0);
```

Comment:
what was the reason to drop "explicit" ?

Reply:
lack of knowledge on my part, i will have to put it back in place... there is a case where it is useful...

Reply:
done